msgpack/src/main/scala/fs2/data/msgpack/low/internal/ItemSerializer.scala

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2024 fs2-data Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2
+package data
+package msgpack
+package low
+package internal
+
+import scodec.bits._
+
+private[low] object ItemSerializer {
+  class MalformedItemError extends Error("item exceeds the maximum size of it's format")
+  class MalformedStringError extends MalformedItemError
+  class MalformedBinError extends MalformedItemError
+  class MalformedIntError extends MalformedItemError
+  class MalformedUintError extends MalformedItemError
```

Comment:
Are those mapped somewhere? Because as they're within a `private[low]`, users won't have access to them and can't distinguish what failed. Maybe something we'd want to allow? The CSV module has public exception types.

Reply:
I have decided that having separate exception for each type is a bit too granular. Instead, there's a general `MsgpackMalformedItemException` class since [482bf9e](https://github.com/gnieh/fs2-data/pull/628/commits/482bf9e5fbbe4bf9cf244d1fa2fc0d6b081ed598).